src/Models/Models.jl

```diff
@@ -25,15 +25,15 @@ import Oceananigans.TimeSteppers: reset!
 function reset!(model::AbstractModel)
 
     for field in fields(model)
-        fill!(field, 0.0)
+        fill!(field, zero(model.grid))
```

Comment:
This assumes that `model.grid` exists. `eltype(model)` is probably better.

Reply:
that means we need to define `Base.eltype(model)` first, right? How do we define that without using the model's grid?

Reply:
Yeap, did that!